netty/src/main/java/io/grpc/netty/NettySocketSupport.java

```diff
@@ -52,9 +57,27 @@ public NativeSocketOptions(
         Map<String, String> otherInfo) {
       Preconditions.checkNotNull(otherInfo);
       this.tcpInfo = tcpInfo;
-      this.otherInfo = ImmutableMap.copyOf(new HashMap<String, String>(otherInfo));
+      this.otherInfo = ImmutableMap.copyOf(otherInfo);
     }
   }
 
-  private NettySocketSupport() {}
+  public static NettySocketSupport instance() {
```

Comment:
this will be wrong for subclasses right?

Reply:
Restructured to avoid ambiguity